### PR TITLE
Update code to run on .NET 5 with the latest Azure SDK

### PR DESCRIPTION
**Pull request by justinyoo**
This is the PR based on the Azure DevOps [item](https://dev.azure.com/devrel/Cloud%20Advocacy/_workitems/edit/33864)

After merging this PR, the code will:

* Run on .NET 5
* Use the latest Azure SDK for [Blob Storage](https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/storage/Azure.Storage.Blobs)
* Change scenario to copy over modified blobs to the destination storage.

This is the draft PR and I'm about to start working on the [Learn module unit](https://github.com/MicrosoftDocs/learn-pr/blob/master/learn-pr/azure/copy-blobs-from-command-line-and-code/includes/8-exercise-move-blobs-using-net-storage-client.md).

### CODE REVIEW

**Comment by justinyoo** [comment]
@adrianstevens @patridge Who can review this PR?

**Review by patridge — COMMENTED** [review]
Did we want the VS Code configuration files included here or just the actual code/project changes?

**Comment by justinyoo** [comment]
Oops! I'll remove the .vscode part.

**Comment by justinyoo** [comment]
.vscode removed

**Comment by justinyoo** [comment]
I also restored the existing sample codes because it should exist until the main learn module is refreshed.

File: code/TransferBlobs/Program.cs [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+using Azure;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+using Azure.Storage.Sas;
+
+namespace TransferBlobs
+{
+    class Program
+    {
+        static void Main(string[] args)
```
Comment (patridge) [inline code/TransferBlobs/Program.cs]: Not sure how much it matters here, but do we want to use `async` main (C# 7.1+) so we can properly await `MoveMatchingBlobsAsync` on line 39?

**Review by patridge — APPROVED** [review]
One comment on async/await, but it looks good to me regardless.

File: code/TransferBlobs/Program.cs (hunk 1 above)
Comment (justinyoo) [reply code/TransferBlobs/Program.cs]: Yeah, makes sense. I've updated it.

**Pull request merged**